plugins/kdeconnect

```
@@ -22,16 +22,29 @@ note() {
 }
 
 send() {
-	xargs -0 -I{} kdeconnect-cli --name "$devname" --share {}
+	xargs -0 -I{} sh -c 'if [ ! -d "{}" ]; then printf "%s\0" "{}"; fi;' |  xargs -0 -I{} kdeconnect-cli --name "$devname" --share {}
```

What's this about?

---

if you pass a directory, kndeconnect creates an empty file on the device with the name of the directory. Basically it doesn't support directory transfers.

---

As far as I see, this simply filters out directories, right? It doesn't recursively send all the files inside the dir?

If that's the case, then just leave it out. The user can simply not select a directory. And if he does select one, it's better to see the empty file than to silently discard his selection.

---

Yes it just filters out directory.

1. But I think we will need some kind of checks. What happens if you send over a block device?
2. In general, how can I have xargs call a user-defined function? On SO [I found I can export the function](https://stackoverflow.com/a/11003457/1719931), but `shellcheck` complains that is not POSIX sh-compatible.

---

it hangs

---

In that case, filtering out only the files seems like a good idea. Does this work?

```diff
diff --git a/plugins/kdeconnect b/plugins/kdeconnect
index 0d3ef8b7..209bf137 100755
--- a/plugins/kdeconnect
+++ b/plugins/kdeconnect
@@ -21,8 +21,18 @@ note() {
 	fi
 }
 
+# kdeconnect doesn't cope with non-files
+filter_files() {
+	xargs -0 -I{} sh -c \
+		'if [ -f "{}" ]; then'\
+		'	printf "%s\0" "{}";'\
+		'else'\
+		'	printf "Error: not a file: %s\n" "{}" >&2;'\
+		'fi;'
+}
+
 send() {
-	xargs -0 -I{} kdeconnect-cli --name "$devname" --share {}
+	filter_files | xargs -0 -I{} kdeconnect-cli --name "$devname" --share {}
 	note "Files sent"
 }
 
```

---

No it says:
```
Pick a device: 1
	printf "%s\0" "src";: -c: line 2: syntax error: unexpected end of file
	printf "%s\0" "CHANGELOG";: -c: line 2: syntax error: unexpected end of file
	printf "%s\0" "LICENSE";: -c: line 2: syntax error: unexpected end of file
```

but I made it to work by removing the `\` characters :)

---

How can we call:

```
note "Error: not a file:
```

instead of:

```
printf "Error: not a file:
```
?

---

Don't think that's possible without doing something dirty. So I'd leave it be.